Frontend/CO.CDP.OrganisationApp/Pages/Registration/OrganisationDetailsSummary.cshtml.cs

```diff
@@ -42,22 +43,29 @@ public async Task<IActionResult> OnPost()
 
     private async Task<OrganisationWebApiClient.Organisation?> RegisterOrganisation(RegistrationDetails details)
     {
-        try
+        var payload = NewOrganisationPayload(details);
+
+        if (payload is not null)
         {
-            var payload = NewOrganisationPayload(details);
-            if (payload is not null)
+            try
             {
                 return await organisationClient.CreateOrganisationAsync(payload);
             }
-        }
-        catch (ApiException aex)
-            when (aex is ApiException<OrganisationWebApiClient.ProblemDetails> pd)
-        {
-            Error = pd.Result.Detail;
-        }
-        catch (Exception ex)
-        {
-            Error = ex.Message;
+            catch (ApiException<OrganisationWebApiClient.ProblemDetails> aex)
```

Comment:
Don't we need to show error page, when there is 500 error from api.
Also code can be shorten too - 

```
if (payload is not null)
{
    try
    {
        return await organisationClient.CreateOrganisationAsync(payload);
    }
    catch (ApiException<OrganisationWebApiClient.ProblemDetails> aex)
        when (aex.StatusCode == StatusCodes.Status400BadRequest)
    {
        ModelState.AddModelError(string.Empty, ErrorMessagesList.OrganisationCreationFailed);
    }
}
```

Reply:
Thanks Updated this.